service/src/main/java/com/google/auto/service/processor/AutoServiceProcessor.java

```diff
@@ -127,26 +131,32 @@ private void processAnnotations(Set<? extends TypeElement> annotations,
     for (Element e : elements) {
       // TODO(gak): check for error trees?
       TypeElement providerImplementer = (TypeElement) e;
-      AnnotationMirror providerAnnotation = getAnnotationMirror(e, AutoService.class).get();
-      DeclaredType providerInterface = getProviderInterface(providerAnnotation);
-      TypeElement providerType = (TypeElement) providerInterface.asElement();
-
-      log("provider interface: " + providerType.getQualifiedName());
-      log("provider implementer: " + providerImplementer.getQualifiedName());
-
-      if (!checkImplementer(providerImplementer, providerType)) {
-        String message = "ServiceProviders must implement their service provider interface. "
-            + providerImplementer.getQualifiedName() + " does not implement "
-            + providerType.getQualifiedName();
-        error(message, e, providerAnnotation);
-      }
+      AnnotationMirror annotationMirror = getAnnotationMirror(e, AutoService.class)
```

Comment:
I think you can revert this line change, the comment doesn't do a whole lot.

Reply:
ac5a649